pkg/controller/nodemaintenance/nodemaintenance_controller.go

```diff
@@ -47,7 +53,7 @@ func onPodDeletedOrEvicted(pod *corev1.Pod, usingEviction bool) {
 	klog.Info(msg)
 }
 
-func initDrainer(r *ReconcileNodeMaintenance, config *rest.Config) error {
+func initDrainer(r *ReconcileNodeMaintenance,cs kubernetes.Interface) error {
```

Comment:
We can revert this, it is unnecessary

Reply:
I had a problem: the initDrainer is creating a real client-go instance based on the rest.Config passed as parameter, whereas the unit tests are using a fake client. I had to force it both to use the same client during the unit test.

Reply:
As discussed on Thursday, without the duplicated e2e test there is no need to pass a fake client to `checkLeaseSupported()` via `initDrainer()`.